lib/Reconciler.lua

```diff
@@ -156,115 +120,118 @@ end
 	the reconciliation methods; they depend on this structure being well-formed.
 ]]
 function Reconciler._mountInternal(element, parent, key, context)
-	if isPrimitiveElement(element) then
-		-- Primitive elements are backed directly by Roblox Instances.
+	if type(element) == "table" then
```

Comment:
Same here, probably not worth checking if the element is a table; we can do typechecking assertions in `mount` though in a later PR, since it's a user-facing API.